### ISSUE TITLE
Rule Change: Add support for TypeScript private members in `no-unused-private-class-members`

### ISSUE THREAD

**Issue by sethamus** [eslint/eslint#20488]
### What rule do you want to change?

no-unused-private-class-members

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
class A {
  private foo = 123;
}
```

### What does the rule currently do for this code?

No warnings for private members declared with the TypeScript "private" keyword (only #private is handled).

### What will the rule do after it's changed?

Report unused members declared with the TypeScript "private" keyword

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

This would align ESLint core behavior with [@typescript-eslint/no-unused-private-class-members](https://typescript-eslint.io/rules/no-unused-private-class-members/).

**Comment by DMartens** [comment]
I think this issue originated from this [comment](https://github.com/eslint/eslint/issues/19173#issuecomment-3798655603) on the issue tracking TypeScript-syntax support in the core rules.
The rule author mentions there that it is harder to detect as it is not as constrained as using "native" private members. [The additional code](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/util/class-scope-analyzer/classScopeAnalyzer.ts) seems to be a lot to handle this.
So I would be a soft no on this proposal.

**Comment by snitin315** [comment]
I agree with @DMartens, this would be hard to maintain. We can revisit later if it is requested by multiple users
